src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php

```diff
@@ -51,20 +51,30 @@ public function init(): int
     private static function preHook($variant)
     {
         return static function (HookData $hook) use ($variant) {
-            if (count($hook->args) == 0 || !is_string($hook->args[0])) {
-                return;
-            }
+           if (count($hook->args) == 0 || !is_string($hook->args[0])) {
+               return;
+           }
+
+           $protocol = [];
+           $uri_parsed = preg_match('/^([a-z]+)\:\/\//', $hook->args[0], $protocol);
```

Comment:
Not a big deal but an alternative could've been `parse_url`: https://www.php.net/manual/en/function.parse-url.php